### DISCUSSION TITLE
New features for the Connect panel

### DISCUSSION THREAD

**Discussion by Zyfuss** [jamulussoftware/jamulus#1490] [UI Related]
Hi, in my opinion would be useful to have these features in the Connect panel:

1. Have a checkbox to select if the panel will be closed or not after the connection to a server. If deselected the panel will be mantained open so to quickly move from one server to another (ie. while performing in Woldjam where you must move quickly from one server to another)
2. Buttons to skip from one server and auto connect to another. While connected to a server these buttons will be used to skip and auto connect to the next or previuos server in the list with at last one (or maybe two? more?) musician. Usually I jump from one server to another finding someone to jam and it takes some clicks. This sort of "auto tuning" like in the car radio could be a fast way to scan the server list.

Thanks. Ciao.

**Comment by gilgongo** [comment]
On point 1, leaving the window open means a lot of network traffic (and possibly UDP packet fragmentation) would be happening while you played, as Jamulus would be polling all servers for ping times and player info. This might degrade the quality of your session. It has been suggested that instead, the connect window should poll only once when opened (regardless of mouse pointer position which currently stops polling while the mouse is over the server list), and we should leave the window open with a "refresh" button there instead. BTW you can in fact leave the window open if you open it from the menu bar.

On point 2, I think that's a new suggestion which is quite interesting.

**Reply by dcorson-ticino-com** [reply]
Hi Zyfuss,
Try out my NewUI draft to see if that solves some of your requests:
https://github.com/dcorson-ticino-com/jamulus/releases/tag/r3_8_0beta3NewUI
Don

**Comment by Zyfuss** [comment]
I did a fast try to this NewUI release and I've to say that is really
interesting.
Most of the information is only one click away.
Also, the Connect panel as is shown now will respond to my observation, the
list is fixed at last update, will be accessible very fast and only updated
on demand.

Thanks. Ciao.

Il giorno dom 11 apr 2021 alle ore 13:08 DonC ***@***.***> ha
scritto:

**Comment by Zyfuss** [comment]
Sorry to be here again... I've a suggestion for the NewUI.
It would be useful to have in the Settings->Audio Setup tab the L-R Input
meter. The one present in the main window is too small to have a precise
setting of the input level from the audio interface.
This is something well checked in the WorldJam soundcheck where your input
level should peak two yellows...

Thanks. Ciao.

Il giorno dom 11 apr 2021 alle ore 14:39 Fulvio Pecorari ***@***.***>
ha scritto:

**Comment by Zyfuss** [comment]
Or... arranging the objects at the top of the main window should be
possible to have an horizontal input meter larger than the one already
there and more usable.

Thanks. Ciao.

Il giorno dom 11 apr 2021 alle ore 14:51 Fulvio Pecorari ***@***.***>
ha scritto:

**Reply by dcorson-ticino-com** [reply]
Thanks for the good ideas.
I will keep them in mind.
This NewUI draft is just a version to test ideas, it is not really a preview of what is coming.
But putting the input balance into the settings is planned, if I am not mistaken.
